primefaces/src/main/resources/META-INF/resources/primefaces/core/core.js

```diff
@@ -102,6 +102,22 @@
             });
         },
 
+        /**
+         * Gets the form by id or the closest form if the id is not a form itself.
+         * @param {string} id ID of the component to get the closest form.
+         * @return {JQuery} the form or NULL if no form found
+         */
+        getForm: function(id) {
```

Comment:
we should name it "getClosestForm"

in AJAX we also have a fallback for the first form in DOM, this should not be used here